src/BlazorWebView/src/WebView2/Internal/StaticContentProvider.cs

```diff
@@ -0,0 +1,599 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.WebView.WebView2.Tmp;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore.Components.WebView.WebView2.Internal
+{
+	internal class StaticContentProvider
+	{
+		private readonly IFileProvider _fileProvider;
+		private readonly Uri _appBaseUri;
+		private readonly string _hostPageRelativePath;
+		private static readonly FileExtensionContentTypeProvider ContentTypeProvider = new();
+		private static readonly ManifestEmbeddedFileProvider _manifestProvider =
+			new ManifestEmbeddedFileProvider(typeof(WebViewManager).Assembly);
+
+		public static StaticContentProvider ResolveFromStaticWebAssetsManifest(IFileProvider fileProvider, Uri uri, string hostPageRelativePath)
+		{
+			var assembly = Assembly.GetEntryAssembly();
+			if (string.IsNullOrEmpty(assembly?.Location))
```

Comment:
I am not sure how this works on mobile things, but I don't think it works on Android.
If this is not for mobile but for desktop, then this is fine.

Reply:
This is only for WPF and WinForms, but thanks for looking at it!